### DISCUSSION TITLE
what is JavaScriptCore?

### DISCUSSION THREAD

**Discussion by jimmywarting** [oven-sh/bun#1037] [Q&A] [answered]
is JavaScriptCore the same engine used by safari?

**Comment by skube** [comment, marked as answer]
https://en.wikipedia.org/wiki/WebKit#JavaScriptCore

Apparently so.

**Reply by jimmywarting** [reply]
hmm, so that's why bun.js is so much faster than deno and nodejs
i have always liked safari for it's speed. but never really liked safari or barly used it b/c it lacks so many new hot features.

**Reply by xhyrom** [reply]
bun is fast because JavaScriptCore and Zig

**Reply by jimmywarting** [reply]
👍 all props to you guys!

Sitting on a mac myself and running NodeJS & chrome/v8 all day long as a web developer. love it for it's bleding edge functionallity and great dev tools but don't use safari b/c it lacks things such as file system access, web bluetooth nfc, web push and what not and the dev tool is kind of B in regards to chromes.

safari is really much more faster and snappier than chrome and also uses less cpu/memory but it also lacks many new cool web features :(  
really wish safari could take torch and wrap up the release cycle and not ship new updates only when there is a new OS update. 
i like safari and i really want to use it instead of the spying google chrome browser

I'm reflecting back now to when chrome forked webkit and started developing chromium and v8 and wonder what it could have been instead. 
we would not have the brave browser or Edge running chromium. would IE have lived on then? perhaps so...
